File: 1603-design-parking-system/solution.py
# Runtime: 97ms, Beats 88.69% of users with Python3
# Memory: 17.04MB, Beats 40.35% of users with Python3

class ParkingSystem:

    def __init__(self, big: int, medium: int, small: int):
        self.big=big
        self.medium=medium
        self.small=small

    def addCar(self, carType: int) -> bool:
        match carType:
            case 1:
                if self.big:
                    self.big-=1
                    return True
                return False
            case 2:
                if self.medium:
                    self.medium-=1
                    return True
                return False
            case 3:
                if self.small:
                    self.small-=1
                    return True
                return False


# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)

File: 1603-design-parking-system/solution.cpp
// Runtime: 45ms, Beats 66.91% of users with c++
// Memory: 37.08MB, Beats 14.46% of users with c++


class ParkingSystem {
public:
    int big;
    int medium;
    int small;
    ParkingSystem(int big, int medium, int small) {
        this->big = big;
        this->medium = medium;
        this->small = small;
    }
    
    bool addCar(int carType) {
        if(carType == 1) {
            if(this->big) {
                this->big--;
                return true;
            }
            return false;
        }
        else if(carType == 2) {
            if(this->medium) {
                this->medium--;
                return true;
            }
            return false;
        }
        else {
            if(this->small) {
                this->small--;
                return true;
            }
            return false;
        }
    }
};

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem* obj = new ParkingSystem(big, medium, small);
 * bool param_1 = obj->addCar(carType);
 */